### DISCUSSION TITLE
[BUG] WCC has different results between num_workers=1 and num_workers>1 on ldbc with project (person->knows->person))

### DISCUSSION THREAD

**Discussion by hanjf12** [alibaba/GraphScope#2486] [Q&A] [answered]
**Describe the bug**
When run with num_workers=1, the result of wcc is every vertex with its oid. And when num_workers>1 result is fine

**Environment (please complete the following information):**
 - GraphScope version:0.19
 - OS: ubuntu20.04

**To Reproduce**
```import graphscope
from graphscope.client.session import session
graphscope.set_option(show_log=True) 
graphscope.set_option(log_level='DEBUG')
from graphscope.dataset import load_p2p_network, load_ldbc

sess = session(cluster_type="hosts", num_workers=1)

graph = load_ldbc(sess=sess)
graph = graph.project(vertices={"person": []}, edges={"knows": []})
wcc_context = graphscope.wcc(graph)
wcc_context.to_dataframe(
    selector={"id": "v.id", "cc": "r"}
).sort_values(by="id")

**Comment by sighingnow** [comment, marked as answer]
WCC only works on undirected graphs. You need

```python
graph = load_ldbc(sess=sess, directed=False)
```

to make it work as expected.

**Reply by sighingnow** [reply]
The error message will be improved in https://github.com/alibaba/GraphScope/pull/2487

**Reply by hanjf12** [reply]
> WCC only works on undirected graphs. You need
> 
> ```python
> graph = load_ldbc(sess=sess, directed=False)
> ```
> 
> to make it work as expected.

Thanks!!

**Comment by sighingnow** [comment]
Closing as answered.
